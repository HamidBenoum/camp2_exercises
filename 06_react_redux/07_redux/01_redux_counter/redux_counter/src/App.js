import React, { Component } from 'react';
import { connect } from "react-redux";
import './App.css';

class App extends Component {
  render() {
    return (
      <div className="App">
          <div>
          <button onClick={ this.props.decrement }> - </button>
          <span> {this.props.counter} </span>
          <button onClick={ this.props.increment }> + </button>
          </div>
      </div>
    );
  }
}

function mapStateToProps(state) {
  return { counter: state }
}

function mapDispatchToProps(dispatch) {
  return {
    decrement: () => dispatch({type: "DECREMENT"}),
    increment: () => dispatch({type: "INCREMENT"})
  }
}

const ConnectedApp = connect(mapStateToProps, mapDispatchToProps)(App);

export default ConnectedApp;
