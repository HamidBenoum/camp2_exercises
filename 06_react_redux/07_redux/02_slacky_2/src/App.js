import React, { Component } from "react";
import "./App.css";
import {ConnectedLogin} from "./components/LoginRedux";
import Chat from "./components/ChatRedux";
import Store from "./components/store";

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      userName: null,
      messages: [],
    };
    // Attaching the websocket to our App so we can reuse it
    this.websocket = new WebSocket("ws://localhost:8080");
  }
  componentDidMount() {
    // Listen for messages
    this.websocket.addEventListener("message", event => {
      const message = JSON.parse(event.data);
      console.log("Message from server ", message);
      switch (message.type) {
        case "CONNECTION_START":
        default:
          return;
        case "MESSAGES":
          this.setState({ messages: message.data });
          return;
      }
    });
  }

  handleUserName = userName => {
    this.setState({ userName: userName });
    this.websocket.send(
      JSON.stringify({
        type: "LOGIN",
        userName: userName
      })
    );
  };

  sendMessage = message => {
    this.websocket.send(
      JSON.stringify({
        type: "NEW_MESSAGE",
        userName: this.state.userName,
        message: message
      })
    );
  };

  render() {
    return (
      <div className="App">
        <header className="App-header">
          <h1 className="App-title">Slacky Redux</h1>
        </header>

        {this.state.userName ? (
          <Chat sendMessage={this.sendMessage} messages={this.state.messages} />
        ) : (
          <ConnectedLogin />
        )}
      </div>
    );
  }
}

export default App;
