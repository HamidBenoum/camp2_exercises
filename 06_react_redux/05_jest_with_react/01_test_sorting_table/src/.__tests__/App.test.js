import React, { Component } from "react";
import App from '../App.js';
import renderer from 'react-test-renderer';
import {shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Wapiti from 'wapiti';

beforeAll(() => {
  configure({ adapter: new Adapter() });
})

const fakeData = [
  {decathlon_id: 2, title: "chaufferette", price: 5.99},
  {decathlon_id: 3, title: "pantalon", price: 14.99},
  {decathlon_id: 1, title: "ekiden", price: 21.99}
];


test('expect array ', () => {
  expect.assertions(1);
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});


test("Should order the list by the title", () => {
  expect.assertions(1);
  const load = shallow(<App lines={fakeData}/>);
  load.find("thead tr th").at(1).simulate("click");
  expect(load.find("tr").at(1).text()).toBe("2chaufferette5.99");
})


//Test sur le navigateur en live.
//test sorting table
test("should sort the array when clicking on the first Column", () => {
  expect.assertions(1)
  return Wapiti.goto("http://localhost:3000")
    .capture(() => document.querySelector("td").textContent)
    .click("th")
    .capture(() => document.querySelector("td").textContent)
    .click("th")
    .capture(() => document.querySelector("td").textContent)
    .run()
    .then(result => {
      expect(result).toEqual(["8044622","8380024", "8044622"]);
    });
});
