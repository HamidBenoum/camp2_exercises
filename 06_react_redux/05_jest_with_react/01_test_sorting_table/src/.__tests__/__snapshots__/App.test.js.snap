// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect array  1`] = `
<table>
  <thead>
    <tr>
      <th
        onClick={[Function]}
      >
        ID
      </th>
      <th
        onClick={[Function]}
      >
        Title
      </th>
      <th
        onClick={[Function]}
      >
        Price
      </th>
    </tr>
  </thead>
</table>
`;
